crates/buffrs/Cargo.toml

```diff
@@ -13,64 +13,30 @@ readme.workspace = true
 repository.workspace = true
 version.workspace = true
 
-[[bin]]
-name = "buffrs"
-path = "src/main.rs"
-required-features = ["git", "validation"]
-
 [[test]]
 name = "e2e"
 path = "tests/lib.rs"
 test = true
 
-[features]
-default = ["git", "validation"]
-validation = ["dep:anyhow", "dep:protobuf", "dep:protobuf-parse", "dep:diff-struct"]
-git = []
-
-
 [dependencies]
-async-recursion.workspace = true
-bytes.workspace = true
+buffrs-core = { workspace = true, features = ["git", "validation"] }
+
 clap.workspace = true
-flate2.workspace = true
-hex.workspace = true
-home.workspace = true
 human-panic.workspace = true
 miette.workspace = true
-reqwest.workspace = true
 semver.workspace = true
-serde.workspace = true
-serde_json.workspace = true
-tar.workspace = true
-thiserror.workspace = true
 tokio.workspace = true
```

Comment:
So my hint here was basically, we will save something by this layout! But it's mostly tokio and clap and I think this save can be achieved with less noise (e.g. having to republish crates etc.)

Also bare in mind that we solved most of the annoyances of the original issue but it was not properly documented (which is my bad). Eg. we removed the dependencies on openssl and libgit.